### DISCUSSION TITLE
How do I setup helix to use C++ 23 for lsp?

### DISCUSSION THREAD

**Discussion by IshanKBG** [helix-editor/helix#11132] [Q&A]
So I use helix on MacOS and wanted to ask how does one set clangd to use a specific C++ version.
Also another issue I am facing is that instead of using llvm installed by homebrew it is using the apple clang which is outdated and does not come with C++ 23.

<img width="1710" alt="Screenshot 2024-07-11 at 11 59 28 AM" src="https://github.com/helix-editor/helix/assets/71317082/46adfaf6-f626-42ed-b0f9-d87c1f5901c9">

**Comment by 57ave** [comment]
8 months without nothing i have the same problem hope that someone will respond 💀

**Reply by kirawi** [reply]
You havae to configure LSP settings, or use a `.clangd` file to pass args to clang.

.clangd:
```
CompileFlags:
  Add: [-std=c++17, -Wall, -Wextra]
```

**Comment by TornaxO7** [comment]
I'd recommend to use [bear](https://github.com/rizsotto/Bear) hear which will take care of the paths for you.
